src/analysis/validation.jl

```diff
@@ -2,102 +2,238 @@
+        head_loss_wntr = vec(head_loss_wntr), head_loss_watermodels = vec(head_loss_watermodels) )
+
+    return valve_df
+end
+
+# compare pumps
+function get_pump_dataframe(wm_data,wm_solution,wntr_data,wntr_simulation,pump_id)
+    duration = length(wm_solution["solution"]["nw"])            # number of time steps
+    time_step = wm_data["time_step"]/3600                       # length per time step (hour)
+    pump_name = wm_data["nw"]["1"]["pump"][pump_id]["source_id"][2]
+    pump_obj = wntr_data.get_link(pump_name)
+    status = Array{Float64,1}(undef,duration)
+    flow_wntr = Array{Float64,1}(undef,duration)
+    flow_watermodels = Array{Float64,1}(undef,duration)
+    head_gain_wntr = Array{Float64,1}(undef,duration)
+    head_gain_watermodels = Array{Float64,1}(undef,duration)
+    power_wntr = Array{Float64,1}(undef,duration)               # kW
+    power_watermodels = Array{Float64,1}(undef,duration)        # kW
+    cost_wntr = Array{Float64,1}(undef,duration)                # $
+    cost_watermodels = Array{Float64,1}(undef,duration)         # $
+
+    function compute_pump_power(pump_obj,q,dh,wntr_data)
+        eff_curve = wntr_data.get_curve(pump_obj.efficiency).points
+        if eff_curve != nothing
+            for j in 1:length(eff_curve)
+                if j == length(eff_curve)
+                    eff = eff_curve[j][2]/100 # use the last Q-eff point
+                elseif (q >= eff_curve[j][1]) & (q < eff_curve[j+1][1])
+                    # linear interpolation of the efficiency curve
+                    a1 = (eff_curve[j+1][1]-q)/(eff_curve[j+1][1]-eff_curve[j][1])
+                    a2 = (q-eff_curve[j][1])/(eff_curve[j+1][1]-eff_curve[j][1])
+                    eff = (a1*eff_curve[j][2]+a2*eff_curve[j+1][2])/100 
+                    break
                 end
             end
+        elseif wntr_data.options.energy.global_efficiency != nothing
+            eff = wntr_data.options.energy.global_efficiency/100 # e.g. convert 60 to 60% (0.6)
+        else
+            println("No pump efficiency provided, 75% is used")
+            eff = 0.75
+        end
+        power = 9.81*dh*q/eff # kilowatt (kW)
+        return power
+    end
+
+    for t in 1:duration
+        status[t] = wm_solution["solution"]["nw"][string(t)]["pump"][pump_id]["status"]
+        flow_wntr[t] = wntr_simulation.link["flowrate"][pump_name].values[t]
+        flow_watermodels[t] = wm_solution["solution"]["nw"][string(t)]["pump"][pump_id]["q"]
+        start_node = wntr_data.links._data[pump_name].start_node_name
+        end_node = wntr_data.links._data[pump_name].end_node_name
+        # the head gain here measures the head difference, not necessarily the lift of an operating pump
+        head_gain_wntr[t] = wntr_simulation.node["head"][end_node].values[t]-wntr_simulation.node["head"][start_node].values[t]
```

Comment:
Ensure that if the pump is off, the head gain is set to zero, here.

Reply:
@enzoliuyang94, you multiplied both the wm and wntr solution head gains by the wm status. I think you should use the wntr status for the wntr head gain. This is one way to check that the WNTR network controls were defined properly. You can get it from `wntr_simulation.link["status"]["pmp1"].values[t]`.

Reply:
@jjstickel @tasseff I did what Jonathan suggested, and also create 'status_wntr' and 'status_watermodels' as two separate columns in the dataframe output.

Reply:
LGTM now. Thanks.